docs/source/guides/gpu_serving.rst

```diff
@@ -0,0 +1,442 @@
+    from bentoml.service.artifacts.common import PickleArtifact
+
+    @bentoml.env(pip_packages=['tensorflow', 'scikit-learn', 'pandas'] ,\
+          docker_base_image="bentoml/model-server:0.12.1-py38-gpu")
+    @bentoml.artifacts([KerasModelArtifact('model'), PickleArtifact('tokenizer')])
+    class TensorflowService(bentoml.BentoService):
+
+        @api(input=JsonInput())
+        def predict(self, parsed_json):
+            return self.artifacts.model.predict(input_data)
+
+.. code-block:: python
+
+    # bento_packer.py
+    from bento_svc import TensorflowService
+
+    # OPTIONAL: to remove tf memory limit on our card
+    config.experimental.set_memory_growth(gpu[0], True)
+
+    model = load_model()
+    tokenizer = load_tokenizer()
+
+    bento_svc = TensorflowService()
+    bento_svc.pack('model', model)
+    bento_svc.pack('tokenizer', tokenizer)
+
+    saved_path = bento_svc.save()
+
+
+2.4 PyTorch
+^^^^^^^^^^^
+
+.. note::
+    Since PyTorch bundled CUDNN and NCCL runtime with the python library the *RECOMMENDED* way to run your PyTorch service is to install PyTorch with conda
+    via BentoML `@env <http://localhost:8000/api/bentoml.html#env>`_:
+
+    .. code-block:: python
+
+        @env(conda_dependencies=['pytorch', 'torchtext', 'cudatoolkit=11.1'], conda_channels=['pytorch', 'nvidia'],
+
+PyTorch provides a more pythonic way to define device for our deep learning model. This can be used through training and inference tasks
+
+.. code-block:: python
+
+    import torch
+
+    device = torch.device("cuda" if torch.cuda.is_available() else "cpu")
+
+.. note::
+    PyTorch provides users with **OPTIONAL** ``cuda:{device_id}`` or ``cpu:{device_id}`` to explicitly assign GPU if the vendors contain multiple GPUs or CPUs.
+    For mose use-case "cuda" or "cpu" will dynamically allocate GPU resources and fallback to CPU for you.
+
+However, make sure that in our BentoService definition every tensor that is needed for inference *should be cast to the same device as our our model*, see :ref:`pytorch-impl`.
+
+.. note::
+    All of the above apply to ``transformers``, ``PytorchLightning`` or any other variant of PyTorch deep learning frameworks.
+
+.. _pytorch-impl:
+
+PyTorch Implementation
+""""""""""""""""""""""
+
+.. note::
+    refers to `PyTorch gallery <https://github.com/bentoml/gallery/blob/master/pytorch/news-classification-gpu/news-classification.ipynb>`_ for the complete version.
+
+.. code-block:: python
+
+    # bento_svc.py
+
+    from bentoml import BentoService, api, artifacts, env
+    from bentoml.adapters import JsonInput, JsonOutput
+    from bentoml.frameworks.pytorch import PytorchModelArtifact
+    from bentoml.service.artifacts.pickle import PickleArtifact
+    import torch
+
+    device = torch.device("cuda:0" if torch.cuda.is_available() else "cpu")
+
+    @env(conda_dependencies=['pytorch', 'torchtext', 'cudatoolkit=11.1'], conda_channels=['pytorch', 'nvidia'],
+     requirements_txt_file=None)
```

Comment:
is the `requirements_txt_file=None` required here?

Reply:
not needed. Removed